docs/source/architecture/journal.rst

```diff
@@ -2,476 +2,196 @@
-for the validator. Such critical errors would include missing blocks, bad
-indexes, missing chain reference, incomplete blocks or invalid blocks in the
-store. The BlockStore provides an atomic means to update the store when the
-current fork is changed (the chain head is updated).
-
-The BlockStore is a persistent on-disk store of all Blocks in the current
-chain. When the validator is started, the contents of the BlockStore is trusted
-to be the current state of the world.
-
-All blocks stored here are formally complete. The BlockStore allows blocks to
-be accessed via Block ID. Blocks can also be accessed via Batch ID,
-Transaction ID, or block number; for example, ``get_block_by_batch_id``,
-``get_block_by_transaction_id``, ``get_batch_by_transaction``, or
-``get_block_by_number``.
-
-The BlockStore maintains internal mappings of Transaction-to-Block and
-Batch-to-Block. These may be rebuilt if missing or corrupt. This rebuild should
-be done during startup, and not during the course of normal operation. These
-mappings should be stored in a format that is cached to disk, so they are not
-required to be held in memory at all times. As the blockchain grows, these will
-become quite large.
-
-The BlockStore provides an atomic method for updating the current head of the
-chain. In order for the BlockStore to switch forks, it is provided with a list
-of blocks in the new chain to commit, and a list of blocks in the old chain to
-decommit. These lists are the blocks in each fork back to the common root.
-
-The BlockCache
-==============
-
-The Block Cache holds the working set of blocks for the validator and tracks the
-processing state. This processing state is tracked as valid, invalid, or
-unknown. Valid blocks are blocks that have been proven to be valid by the
-ChainController. Invalid blocks are blocks that failed validation or have an
-invalid block as a predecessor. Unknown are blocks that have not yet completed
-validation, usually having just arrived from the Completer.
+- The **BlockCache** and **BlockStore** provide storage for the batches and
+  blocks being processed.
 
-The BlockCache is an in-memory construct. It is rebuilt by demand when the
-system is started.
+Batch and block processing is designed to be asynchronous, which allows the
+ChainController to process incoming blocks in parallel, as well as allowing the
+BlockPublisher to proceed with claiming blocks even when the incoming block rate
+is high.
 
-If a block is not present in the BlockCache, it will look in the BlockStore for
-the block. If it is not found or the lookup fails, the block is unknown to the
-system. If the block is found in the BlockStore it is loaded into the
-BlockCache and marked as valid. All blocks in the BlockStore are considered
-valid.
+This approach is flexible enough to work with different consensus algorithms.
+Sawtooth includes a *consensus interface* that communicates the components
+in the journal.
```

Comment:
The last part of this sentence doesn't make sense. Maybe you meant "that defines communication between the consensus and components in the journal."

Reply:
Oops - I dropped a word by mistake. I meant "that communicates WITH the components in the journal."  I'm trying to avoid any hints of the upcoming consensus engine/consensus proxy for now.